### ISSUE TITLE
Custom Event for Displaying Quest

### ISSUE THREAD

**Issue by cwingrav** [CitizensDev/Citizens#207]
It would be interesting if we could listen in for people about to get a quest. Listing the quests as text just doesn't break from player's normal game time. I want to listen in, capture the even, create a popup w/ spout and list the quest text.

Probably put this in the Quester::updateDescription you could put something that fires off to its listeners and if any return true, then don't do its own display?

**Comment by fullwall** [comment]
There are a couple of events - QuestBeginEvent, QuestCancelEvent and QuestCompleteEvent - did you need any more?

**Comment by czorn** [comment]
How do you hook into them? I'm getting this error with this code:

Could not pass event CUSTOM_EVENT to QuestGUI
java.lang.NoClassDefFoundError: net/citizensnpcs/questers/api/events/QuestBeginEvent

In my CustomEventListener

```
public void onCustomEvent(Event event) {
    super.onCustomEvent(event);

    if(event instanceof QuestBeginEvent) {
        System.out.println("Quest Begin");
    }
}
```

Also, are any events fired when players make progress in a quest? (e.g. picking up their 5th of 10 cobblestone blocks) What event would I want to listen for?

**Comment by fullwall** [comment]
There aren't any progress events yet, but I will add when I get time. A noclassdeffounderror sounds like you're building against a different version of Citizens to the one on your server.

**Comment by cwingrav** [comment]
For some reason, it isn't finding the jar files that you are loading dynamically in your CitizensNPCLoader. However, since we specifically add them to the classpath in Eclipse, it sees them there. Its a weird error.

Basically, we'd just like to have a nice little GUI that pops up when they get quests and complete them.

**Comment by czorn** [comment]
Thanks for the response, I don't think that's the case - I've linked to the jars in the plugin folder for craftbukkit, so they are the same jars the server is using.

The weirdest thing is happening. I'm doing your typical custom event listener for bukkit, and I am being forwarded the Quest events. i.e. If I print out the name and classpath of each event my listener gets passed, I will print out "QuestBeginEvent net.citizensnpcs.questers.api.events.QuestBeginEvent", so the runtime is aware of the Quest classes.

But as soon as I try to cast the events as QuestEvents so that I can get the Quest object from them, I get a noclassdeffounderror regarding the classes in the Quester API.

So in short, the JRE has loaded the Quest related classes (I know this because I am passed Quest events), but when I try to use the Quest related classes in my code, the JRE can't resolve them (I know this because whenever I use one of the classes, the NoClassDefFoundError is thrown).

In a related note, I can access everything from the Core jar for Citizens, just not anything from the Quester jar. I imagine that this is the case with the other jars in the 'types' folder, which leads me to conclude that the problems I am having are caused by the jar being in the separate folder.

Any of your thoughts would be appreciated.

Regards,

**Comment by czorn** [comment]
I had meant to add the code I'm using. 
Here's the Plugin file http://pastebin.com/eYRRLthy
And the Listener http://pastebin.com/WWBp8kH7

In case it's not clear, this is the location of all of the jars

../craftbukkit/plugins/Citizens.jar
../craftbukkit/plugins/Citizens/types/Blacksmith.jar
../craftbukkit/plugins/Citizens/types/Guard.jar
../craftbukkit/plugins/Citizens/types/Healer.jar
../craftbukkit/plugins/Citizens/types/Quester.jar
../craftbukkit/plugins/Citizens/types/Trader.jar
../craftbukkit/plugins/Citizens/types/Wizard.jar

I tried copying the Quester.jar to ../craftbukkit/plugins/Quester.jar, but that didn't solve anything.

Thanks again,
